### ISSUE TITLE
Spawn not working on iOS 13.3.1 with Frida 14.2.2

### ISSUE THREAD

**Issue by tr4px** [frida/frida#1582]
Invoking e.g. `frida -U -f com.apple.stocks` raises a timeout error:

`Failed to spawn: unexpectedly timed out while waiting for app to launch`

The app is spawned successfully but error is displayed after a few seconds. It seems to happen with any app. Furthermore, `--no-pause` does not have an effect. Attaching to an already running app works without any issues though.

I have also tried to look into the issue further based on https://github.com/frida/frida/issues/415#issuecomment-360962255. Therefore, I get the following output when running `frida -U -l launchd.js -p 1 --no-pause`:

<pre>
----- Debug -----
path = /var/containers/Bundle/Application/76680515-3149-4AB0-A280-EA1568D704FE/Stocks.app/Stocks
rawIdentifier = null

----- Debug -----
path = /usr/libexec/xpcproxy
rawIdentifier = com.apple.SCHelper

...
</pre>

I guess the issue is related to `rawIdentifier = null`. I can dedicate some time for digging into this in case no 13.3 device is available for reproducing this issue.

I am on a jailbroken (checkra1n) iPhone 7 with iOS 13.3.1 using Frida 14.2.2

**Comment by 0xXA** [comment]
Hi i have the same problem on android 10 arm64, app spawns but frida from computer can't connect to the device ? @oleavr

**Comment by oleavr** [comment]
@tr4px Interesting! This means we need to handle cases where `path` is not `/usr/libexec/xpcproxy`, i.e. the app is launched directly by launchd. This seems easy except for figuring out its bundle ID, but maybe we can get that from the environment string vector or one of the arguments.

@0xXA This error message is very generic and means a lot of things. So if you're getting that error in an Android setup it could be anything really. My guess is that your system has USAP enabled, which we don't yet support. (So you need to disable it, or contribute support for it.)

**Comment by tr4px** [comment]
@oleavr Yes, exactly. Thanks to https://github.com/frida/frida/issues/415#issuecomment-360962255, I was able to find and fix the issue in frida-core `launchd.js` a few days ago by extracting the bundle ID by calling `tryParseXpcServiceName` as you mentioned. So I basically just check again if `rawIdentifier === null` at [launchd.js#L57](https://github.com/frida/frida-core/blob/master/src/darwin/agent/launchd.js#L57):

<pre>let rawIdentifier = null;
if (path === '/usr/libexec/xpcproxy') {
  rawIdentifier = args[3].add(pointerSize).readPointer().readUtf8String();
}
if (rawIdentifier === null){
  rawIdentifier = tryParseXpcServiceName(args[4]);
  if (rawIdentifier === null)
    return;
}</pre>

I did not have time yet to commit a fix, so maybe you could do that. I actually fixed it in my setup the hacky way by just patching `frida-server` on iOS. I can confirm that this worked for iOS 13.3.1.

**Comment by 0xXA** [comment]
> My guess is that your Android system has USAP enabled
Android 10 has USAP disabled by default

**Comment by Khalid-noor** [comment]
is the issue fixed ? I'm facing the same issue..

**Comment by tr4px** [comment]
I updated Frida to 14.2.11 and I don't get the issue anymore.

However, the proposed change in https://github.com/frida/frida/issues/1582#issuecomment-756350342 was not applied. So it seems something else solved the issue.

**Comment by Khalid-noor** [comment]
I updated to 14.2.11, and I still face this issue..

**Comment by Sn0wfreezeDev** [comment]
I get the same issue with Frida 14.2.13, which seems to be the most current version on pip. 
I tried ` frida-trace -U  -f <appname>` and I tried the using python directly by calling `device.spawn(appid)`. The app launches correctly, but Frida is not able to attach and times out shortly after. 

I am using iOS 14.1 with the unc0ver jailbreak.

**Comment by tr4px** [comment]
I actually got the issue again a few weeks ago. Even applying this https://github.com/frida/frida/issues/1582#issuecomment-756350342 again did not work. I also tried different Frida versions. But after rebooting the phone, the issue disappeared.

**Comment by theart42** [comment]
I am encountering this problem as well. Installed Frida 14.2.18 via Cydia, running on an iPhone 6S, iOS 13.6.1. Spawn times out, and attaching to the running process works fine. I need to overload a jailbreak detection that is run when the app starts.  Any suggestions to fix would be very much appreciated (BTW, tried to install the gadget via objection, but the signer throws a fatal error)

**Comment by dazuiba** [comment]
> I get the same issue with Frida 14.2.13, which seems to be the most current version on pip. I tried ` frida-trace -U -f <appname>` and I tried the using python directly by calling `device.spawn(appid)`. The app launches correctly, but Frida is not able to attach and times out shortly after.
> 
> I am using iOS 14.1 with the unc0ver jailbreak.

I am encountering the same issue.  my I ask anyone have a workaround?

**Comment by dazuiba** [comment]
Re jailbreak my iPhone, it works now. problem solved!
